### ISSUE TITLE
Make LanguageParser independent from IsScript

### ISSUE THREAD

**Issue by ivanbasov** [dotnet/roslyn#17811]
Interactive should be responsible for filtering diagnostics. LanguageParser should perform parsing no matter if Interactive or other consumers called it. 
Consider removing IsScript from LanguageParser. See what should be done on the Interactive side to support this.

**Comment by AlekseyTs** [comment]
I guess this issue is specifically about conditional logic in 
```
        private ExpressionStatementSyntax ParseExpressionStatement(ExpressionSyntax expression)
        {
            SyntaxToken semicolon;
            if (IsScript && this.CurrentToken.Kind == SyntaxKind.EndOfFileToken)
            {
                semicolon = SyntaxFactory.MissingToken(SyntaxKind.SemicolonToken);
            }
            else
            {
                // Do not report an error if the expression is not a statement expression.
                // The error is reported in semantic analysis.
                semicolon = this.EatToken(SyntaxKind.SemicolonToken);
            }

            return _syntaxFactory.ExpressionStatement(expression, semicolon);
        }
```

We probably won't be able to eliminate the logic around accepting global statements.

**Comment by ivanbasov** [comment]
I found 7 references to IsScript within LanguageParser, and I am adding a new one with fixing the code @AlekseyTs quoted.  I thought about cleaning LanguageParser from IsScript completely. Then, Scripting (out of parser) should be responsible for all decisions made for IsScript condition.

**Comment by AlekseyTs** [comment]
> Then, Scripting (out of parser) should be responsible for all decisions made for IsScript condition.

That could be too ambitious.
